diff --git a/recipe/transfer_queue/ray_trainer.py b/recipe/transfer_queue/ray_trainer.py
index 6fe12f2e..a61f6e04 100644
--- a/recipe/transfer_queue/ray_trainer.py
+++ b/recipe/transfer_queue/ray_trainer.py
@@ -1175,7 +1175,7 @@ class RayPPOTrainer:
         global_seqlen_lst = data["attention_mask"].view(batch_size, -1).sum(-1).tolist()  # (train_batch_size,)
         world_size = self.actor_rollout_wg.world_size
         sequence_packing_config = None
-        if hasattr(self.config.actor_rollout_ref.actor, 'sequence_packing'):
+        if self.config.actor_rollout_ref.actor.sequence_packing:
             sequence_packing_config = self.config.actor_rollout_ref.actor.sequence_packing
         packing_algorithm = sequence_packing_config.get("algorithm", PackingAlgorithm.KARMARKAR_KARP) if sequence_packing_config else PackingAlgorithm.KARMARKAR_KARP
         
