diff --git a/verl/trainer/ppo/ray_trainer.py b/verl/trainer/ppo/ray_trainer.py
index 9ced163d..0abbb0e9 100644
--- a/verl/trainer/ppo/ray_trainer.py
+++ b/verl/trainer/ppo/ray_trainer.py
@@ -918,7 +918,7 @@ class RayPPOTrainer:
         global_seqlen_lst = batch.batch["attention_mask"].view(batch_size, -1).sum(-1).tolist()  # (train_batch_size,)
         world_size = self.actor_rollout_wg.world_size
         sequence_packing_config = None
-        if hasattr(self.config.actor_rollout_ref.actor, 'sequence_packing'):
+        if self.config.actor_rollout_ref.actor.sequence_packing:
             sequence_packing_config = self.config.actor_rollout_ref.actor.sequence_packing
         packing_algorithm = sequence_packing_config.get("algorithm", PackingAlgorithm.KARMARKAR_KARP) if sequence_packing_config else PackingAlgorithm.KARMARKAR_KARP
         
